### PR TITLE
Slightly adjust the signature of query_map

### PR DESCRIPTION
**Pull request by marcusklaas**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -636,17 +639,20 @@ impl<'conn> SqliteStatement<'conn> {
         Ok(SqliteRows::new(self))
     }
 
-    pub fn query_map<'a, 'map, T, F>(&'a mut self, params: &[&ToSql], f: F)
-                                     -> SqliteResult<MappedRows<'a, 'map, T>>
+    /// Executes the prepared statement and maps a function over the resulting
+    /// rows. 
+    ///
+    /// Unlike the iterator produced by `query`, the returned iterator will
+    /// never panic.
```
Comment (jgallagher) [inline src/lib.rs]: This claim is too strong - `f` might panic itself, for example. Maybe "Unlike the iterator produced by `query`, the returned iterator does not expose the possibility for accessing stale rows." ?

File: src/lib.rs [hunk 2]
```diff
@@ -700,16 +706,19 @@ impl<'conn> Drop for SqliteStatement<'conn> {
     }
 }
 
-pub struct MappedRows<'stmt, 'map, T> {
+/// An iterator over the mapped resulting rows of a query which never panics.
```
Comment (jgallagher) [inline src/lib.rs]: Here we could probably just drop "which never panics".

File: src/lib.rs (hunk 1 above)
Comment (marcusklaas) [inline src/lib.rs]: That's much better. Will change.

**Comment by marcusklaas** [comment]
Updated the comments

**Pull request merged**